compiler/src/dmd/dtemplate.d

```diff
@@ -4556,9 +4556,15 @@ extern (C++) class TemplateInstance : ScopeDsymbol
 
         // The arguments are not treated as part of a default argument,
         // because they are evaluated at compile time.
+        const inCondition = !!(sc.flags & SCOPE.condition);
+        // For master branch: const inCondition = sc.condition;
         sc = sc.push();
         sc.inDefaultArg = false;
 
+        // https://issues.dlang.org/show_bug.cgi?id=24699
+        sc.flags |= SCOPE.condition * inCondition;
```

Comment:
please write this as `if (inCondition) sc.flags |= SCOPE.condition;`.
multiplication of bool and flags is just plain confusing.

Reply:
Doesn't really matter, it's just temporary code standing in until the next stable merge